src/sentry/api/endpoints/project_ownership.py

```diff
@@ -24,7 +24,54 @@ def validate_raw(self, attrs, source):
                 u'Parse error: %r (line %d, column %d)' % (
                     e.expr.name, e.line(), e.column()
                 ))
-        attrs['schema'] = dump_schema(rules)
+
+        schema = dump_schema(rules)
+
+        user_emails = []
+        team_slugs = []
+        for rule in schema['rules']:
+            for owner in rule['owners']:
+                if owner['type'] is 'user':
+                    user_emails.append(owner['identifier'])
+
+                if owner['type'] is 'team':
+                    team_slugs.append(owner['identifier'])
+
+        user_ids = UserEmail.objects.filter(
```

Comment:
You should use the same queries that are used in projectownership.py, or share the code there. Since we do this twice, and your validation is different from the canonical source.

Reply:
crap. wish i saw this  before.

Reply:
thanks